rust_src/src/editfns.rs

```diff
@@ -77,6 +77,21 @@ pub fn eolp() -> LispObject {
     )
 }
 
+/// Return the position of the gap, in the current buffer.
+/// See also `gap-size'.
+#[lisp_fn]
+pub fn gap_position() -> LispObject {
+    let buffer_ref = ThreadState::current_buffer();
+    LispObject::from_natnum(buffer_ref.gap_position() as EmacsInt)
```

Comment:
natnum is a promise to be greater than or equal to zero. Will this always be true?

Reply:
Yes, the original uses `XSETFASTINT`, replaced by [this function](https://github.com/Wilfred/remacs/blob/master/rust_src/src/lisp.rs#L315-L316).

Reply:
:+1: